packages/muelu/src/Transfers/Generic/MueLu_ReplicatePFactory_def.hpp

```diff
@@ -0,0 +1,192 @@
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#ifndef MUELU_REPLICATEPFACTORY_DEF_HPP
+#define MUELU_REPLICATEPFACTORY_DEF_HPP
+
+#include <stdlib.h>
+#include <iomanip>
+
+
+// #include <Teuchos_LAPACK.hpp>
+#include <Teuchos_SerialDenseMatrix.hpp>
+#include <Teuchos_SerialDenseVector.hpp>
+#include <Teuchos_SerialDenseSolver.hpp>
+
+#include <Xpetra_CrsMatrixWrap.hpp>
+#include <Xpetra_ImportFactory.hpp>
+#include <Xpetra_Matrix.hpp>
+#include <Xpetra_MapFactory.hpp>
+#include <Xpetra_MultiVectorFactory.hpp>
+#include <Xpetra_VectorFactory.hpp>
+
+#include <Xpetra_IO.hpp>
+
+#include "MueLu_FactoryManagerBase.hpp"
+#include "MueLu_ReplicatePFactory_decl.hpp"
+
+#include "MueLu_MasterList.hpp"
+#include "MueLu_Monitor.hpp"
+
+namespace MueLu {
+
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  RCP<const ParameterList> ReplicatePFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::GetValidParameterList() const {
+    RCP<ParameterList> validParamList = rcp(new ParameterList());
+    validParamList->setEntry("replicate: npdes",ParameterEntry(1));
+
+    return validParamList;
+  }
+
+  template <class Scalar,class LocalOrdinal, class GlobalOrdinal, class Node>
+  void ReplicatePFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::
+  DeclareInput(Level& fineLevel, Level& /* coarseLevel */) const {
+//    Input(fineLevel, "Psubblock");
+  }
+
+  template <class Scalar,class LocalOrdinal, class GlobalOrdinal, class Node>
+  void ReplicatePFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Build(Level& fineLevel,
+                                Level& coarseLevel) const {
+    return BuildP(fineLevel, coarseLevel);
+  }
+
+  template <class Scalar,class LocalOrdinal, class GlobalOrdinal, class Node>
+  void ReplicatePFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::BuildP(Level& fineLevel,
+                                Level& coarseLevel) const {
+    FactoryMonitor m(*this, "Build", coarseLevel);
+
+    RCP<Matrix>      Psubblock             = coarseLevel.Get< RCP<Matrix> >("Psubblock", NoFactory::get());
+    const ParameterList& pL = GetParameterList();
+printf("before p print\n");
+std::cout << pL << std::endl;
+printf("after p print\n");
```

Comment:
Remove?

Reply:
Oops ... I missed this one and the following one.  Just fixed it. Thanks for catching.